lsp-bridge-diagnostic.el

```diff
@@ -169,13 +182,18 @@ You can set this value with `(2 3 4) if you just need render error diagnostic."
 
 (defun lsp-bridge-diagnostic--render (filepath filehost diagnostics)
   (lsp-bridge--with-file-buffer filepath filehost
-                                (lsp-bridge-diagnostic-hide-overlays)
-
-                                (let ((diagnostic-index 0)
-                                      (diagnostic-number (length diagnostics)))
-                                  (dolist (diagnostic diagnostics)
-                                    (let ((severity (plist-get diagnostic :severity)))
-                                      (unless (member severity lsp-bridge-diagnostic-hide-severities)
+                                (setq-local lsp-bridge-diagnostic-records
+                                            (seq-remove
+                                             (lambda (diagnostic)
+                                               (member (plist-get diagnostic :severity) lsp-bridge-diagnostic-hide-severities))
+                                             diagnostics))
+                                (run-hooks 'lsp-bridge-diagnostic-update-hook)
```

Comment:
Those code only need when `lsp-bridge-diagnostic-enable-overlays` is nil.

Reply:
- I think keep update-hook unconditionally is a better idea to developers, add-hook would be far better than advice-add with -render function to use. 
- The var `lsp-bridge-diagnostic-records` is reused in  render overlays later to avoid filter by hide-severity twice, not totally redundancy.
-  I think save the originial diagnostics data everytime is quite important, so that developers could get much more infomations such as  error/wanring/info code: e.g. [TS2045]  to search for documentations, diagnositc overlays lose these informations. There is a max-number limit (50 by default) for diagnostics, would it slow-down the performance? I don't know, I guess rendering these overlays would take much more time than save values, maybe we need test.

Reply:
The reason why diagnostic information is limited to 50 is because when Emacs encounters thousands of diagnostic messages, drawing so many overlays, Emacs itself will freeze

Reply:
I could add wrong code info to overlay message, would it be better?

Reply:
@manateelazycat final version is pushed. Only add three lines for render function, have a look!